tools/astarte_e2e/lib/astarte_e2e/config/normalized_mail_address.ex

```diff
@@ -0,0 +1,40 @@
+#
+# This file is part of Astarte.
+#
+# Copyright 2020 Ispirata Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+defmodule AstarteE2E.Config.NormalizedMailAddress do
+  use Skogsra.Type
+
+  def cast(value) when is_binary(value) do
+    regexp = ~r/.+[ ]?<.+@.+>/
+
+    if String.match?(value, regexp) do
+      [name, mail] =
+        value
+        |> String.trim(">")
+        |> String.split("<")
+
+      {:ok, {name, mail}}
+    else
+      {:ok, value}
```

Comment:
Since you're already matching with a regex, use this
https://hexdocs.pm/elixir/Regex.html#named_captures/3
so you can match and extract in a single pass
(I also fixed-up the regex a little bit)
```elixir
    regexp = ~r/^(?<name>.+) <(?<email>.+@.+)>$/

    case Regex.named_captures(regexp, value) do
      %{"email" => email, "name" => name} ->
        {:ok, {name, email}}

      _ ->
        {:ok, value}
    end
```